server/src/main/java/io/crate/expression/scalar/ToCharFunction.java

```diff
@@ -0,0 +1,162 @@
+                            signature,
+                            boundSignature,
+                            type,
+                            ToCharFunction::evaluateTimestamp
+                        )
+                );
+            });
+
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.INTERVAL.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature(),
+                DataTypes.STRING.getTypeSignature()
+            ),
+            (signature, boundSignature) ->
+                new ToCharFunction(
+                    signature,
+                    boundSignature,
+                    DataTypes.INTERVAL,
+                    ToCharFunction::evaluateInterval
+                )
+        );
+
+        DataTypes.NUMERIC_PRIMITIVE_TYPES.stream()
+            .forEach(type -> {
+                module.register(
+                    Signature.scalar(
+                        NAME,
+                        type.getTypeSignature(),
+                        DataTypes.STRING.getTypeSignature(),
+                        DataTypes.STRING.getTypeSignature()
+                    ),
+                    (signature, boundSignature) ->
+                        new ToCharFunction(
+                            signature,
+                            boundSignature,
+                            type,
+                            ToCharFunction::evaluateNumber
+                        )
+                );
+            });
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+    private final DataType expressionType;
+    private final BiFunction<Object, String, String> evaluatorFunc;
+
+    public ToCharFunction(Signature signature, Signature boundSignature, DataType expressionType, BiFunction<Object, String, String> evaluatorFunc) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+        this.expressionType = expressionType;
+        this.evaluatorFunc = evaluatorFunc;
+    }
+
+    private static String evaluateTimestamp(Object timestamp, String pattern) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(pattern,
+                                                                  Locale.ENGLISH);
+        Long ts = DataTypes.TIMESTAMPZ.sanitizeValue(timestamp);
+        LocalDateTime dateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(ts),
+                                                         TimeZone.getTimeZone("UTC").toZoneId());
+        return formatter.format(dateTime);
+    }
+
+    private static String evaluateInterval(Object interval, String pattern) {
+        PeriodFormatter formatter = PeriodFormat.getDefault();
+        return formatter.print((Period) interval);
+    }
+
+    private static String evaluateNumber(Object number, String pattern) {
+        DecimalFormat formatter = (DecimalFormat) NumberFormat.getNumberInstance(Locale.ENGLISH);
```

Comment:
can we move `DecimalFormat` and `PeriodFormatter` out of the method scope? we do not need to init them on each evaluation, right?